python/ray/tests/test_global_gc.py

```diff
@@ -216,5 +217,82 @@ def f(self):
         gc.enable()
 
 
+@pytest.mark.timeout(30)
+@pytest.mark.parametrize("support_fork", [True, False])
+def test_long_local_gc(shutdown_only, support_fork):
```

Comment:
let's add a unit test for the thread implementation as well. you can dependency inject a fake `gc.collect` implementation and check that the event, joining, etc. logic works correctly